runtimes/pythonrt/runner/main.py

```
@@ -209,49 +230,43 @@ def Start(self, request: pb.runner.StartRequest, context: grpc.ServicerContext):
 
     def Execute(self, request: pb.runner.ExecuteRequest, context: grpc.ServicerContext):
         with self.lock:
-            call = self.activity_calls[-1] if self.activity_calls else None
+            call: Call = self.activity_call
 
         if call is None:
-            context.abort(grpc.StatusCode.INVALID_ARGUMENT, "no pending activity calls")
+            context.abort(grpc.StatusCode.INTERNAL, "no pending activity calls")
 
-        func_name = full_func_name(call.fn)
-        log.info("calling %s", func_name)
-        result = err = None
+        result = self._call(call.fn, call.args, call.kw)
         try:
-            result = call.fn(*call.args, **call.kw)
-        except Exception as e:
-            log.error("%s failed: %s", func_name, e)
-            display_err(call.fn, e)
-            err = e
+            data = pickle.dumps(result)
+        except (TypeError, pickle.PickleError) as err:
```

what about other errors ?

---

These are the errors pickle raises. The rule-of-thumb in Python is to catch specific exceptions and not `Exception`. If you catch `Exception` all the time you might mask errors you don't know how to handle.